signer/src/service/api/mod.rs

```diff
@@ -72,31 +72,98 @@ fn signer_service_api_inner(command: JsonCommandRequest) -> Result<JsonCommandRe
                 account_info,
             }
         }
-        JsonCommandRequest::get_account { mnemonic } => {
-            let account_info = service::get_account(&mnemonic)?;
-            JsonCommandResponse::get_account { account_info }
-        }
+        JsonCommandRequest::get_account {
+            mnemonic,
+            bip39_entropy,
+        } => match mnemonic {
+            Some(mnemonic) => match bip39_entropy {
+                Some(_) => {
+                    return Err(anyhow!(
+                        "Only one of mnemonic or bip39_entropy can be provided"
+                    ));
+                }
```

Comment:
This could probably be done at a later time, but if we're going to force only one of mnemonic or bip39 entropy then I would suggest that we create an enum for them and handle it at the decoding level so that we don't need to have this logic triplicated

I _think_ it would be something like the following, I'm not fond of the name I chose.
Again we can come back later, and this PR can focus on getting this change out for a fast turnaround.
```rust
#[derive(Deserialize, Serialize, EnumIter, Debug)]
#[serde(tag = "method", content = "params")]
#[allow(non_camel_case_types)]
pub enum Derivation {
    Mnemonic(String),
    Bip39Entropy(String),
}

/// Requests to the Signer Service.
#[derive(Deserialize, Serialize, EnumIter, Debug)]
#[serde(tag = "method", content = "params")]
#[allow(non_camel_case_types)]
pub enum JsonCommandRequest {
    create_account {},
    get_account {
        derivation: Derivation
    },
    sign_tx {
        derivation: Derivation,
        unsigned_tx_proposal: UnsignedTxProposal,
    },
    sync_txos {
        derivation: Derivation,
        txos_unsynced: Vec<TxoUnsynced>,
    },
}
```

Reply:
We should definitly hold off here, thinking about it more I think it will just let the last one win, if both show up in the json.